certbot-ci/certbot_integration_tests/certbot_tests/context.py

```diff
@@ -60,71 +53,24 @@ def cleanup(self):
         """Cleanup the integration test context."""
         shutil.rmtree(self.workspace)
 
-    def _common_test_no_force_renew(self, args):
-        """
-        Base command to execute certbot in a distributed integration test context,
-        not renewing certificates by default.
-        """
-        new_environ = os.environ.copy()
-        new_environ['TMPDIR'] = self.workspace
-
-        additional_args = []
-        if self.certbot_version >= LooseVersion('0.30.0'):
-            additional_args.append('--no-random-sleep-on-renew')
-
-        command = [
-            'certbot',
-            '--server', self.directory_url,
-            '--no-verify-ssl',
-            '--http-01-port', str(self.http_01_port),
-            '--https-port', str(self.tls_alpn_01_port),
-            '--manual-public-ip-logging-ok',
-            '--config-dir', self.config_dir,
-            '--work-dir', os.path.join(self.workspace, 'work'),
-            '--logs-dir', os.path.join(self.workspace, 'logs'),
-            '--non-interactive',
-            '--no-redirect',
-            '--agree-tos',
-            '--register-unsafely-without-email',
-            '--debug',
-            '-vv'
-        ]
-
-        command.extend(args)
-        command.extend(additional_args)
-
-        print('Invoke command:\n{0}'.format(subprocess.list2cmdline(command)))
-        return subprocess.check_output(command, universal_newlines=True,
-                                       cwd=self.workspace, env=new_environ)
-
-    def _common_test(self, args):
+    def _common_test(self, args, force_renew):
         """
-        Base command to execute certbot in a distributed integration test context,
-        renewing certificates by default.
+        Base command to execute certbot in a distributed integration test context.
         """
-        command = ['--renew-by-default']
-        command.extend(args)
-        return self._common_test_no_force_renew(command)
+        return certbot_call.certbot_test(
+            args, self.directory_url, self.http_01_port, self.tls_alpn_01_port,
+            self.config_dir, self.workspace, force_renew, True)
 
-    def certbot_no_force_renew(self, args):
```

Comment:
optional nit: Because calls like:
```
context.certbot(['renew', '--deploy-hook', 'echo deploy >> "{0}"'.format(context.hook_probe)],
                            False)
```
look pretty mysterious when `force_renew` isn't set as a keyword argument, I think I prefer keeping `certbot_no_force_renew` and `certbot` around. You can make the call on that though.

Reply:
Are you talking about a potential weird use here? Because in the current code I always pass the value as a keyword argument. As passing an optional value positionnally is generally not a good practice, I would expect that this kind of usage would be catched during a review if introduced.

My main concern here is to remove the old behavior, inherited from the bash scripts, to have 5 methods to call certbot, and have a situation where `--renew-by-default` then `--no-force-renew` are passed to Certbot when `certbot_no_force_renew` was invoked, and expect things are working as expected because magic.